src/js/kits/model-kit-select/model-kit-select.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { ModelKitConsumer } from '../../data/kits/modelkit';
+import { BrandSelect } from './brand-select';
+import { ScaleSelect } from './scale-select';
+import { SearchInput } from './search-input';
+import { KitPreview } from './kit-preview';
+
+const DEBOUNCE = 300;  // debounce in ms
+
+const modelKitConsumer = new ModelKitConsumer();
+
+// TODO: pass selected kit IDs as prop
+// TODO: handle toggle changes & allowMultiple yes/no
+
+const ModelKitSelect = (props) => {
+    const {
+        label,
+        htmlName,
+        allowMultiple,
+        selected,
+    } = props;
+
+    // track search parameters state
+    const [searchParams, setSearchParams] = useState({});
+    const emptySearch = !Object.keys(searchParams).length;
+    // wrap around new object creation to update the search params, as key modification
+    // is not observed by React. Also clears the search key if the value is empty.
+    const updateSearchParam = (param, value) => {
+        const newSearchParams = Object.assign({}, searchParams, {[param]: value});
+        if (!value) {
+            delete newSearchParams[param];
+        }
+
+        if (param !== 'page') {
+            delete newSearchParams.page;
+            setSearchResults([]);
+        }
+
+        setSearchParams(newSearchParams);
+    }
+
+    const [searchResults, setSearchResults] = useState([]);
+    const [lastResultList, setLastResultList] = useState(null);
+
+    // make an API call whenever the search params change
+    useEffect(() => {
+        if (emptySearch) {
+            return;
+        }
+        const tid = window.setTimeout(
```

Comment:
I see that `useDebounce` returns a cancel function, but I'm not sure how that should be called with a functional component that gets unmounted?

Reply:
Hmm good point, in this case it can be actually better to use `useAsyncFn` and debounce that instead: https://github.com/grafana/grafana/blob/master/public/app/core/components/Select/DashboardPicker.tsx#L42. `useAsyncFn` handles the unmounted state: https://github.com/streamich/react-use/blob/master/src/useAsyncFn.ts#L38

Reply:
right, I see. Not gonna tackle this right now, but when I eventually run into it, I'll know where to look. Currently this component is not used in any dynamic interactions, but there are refactor-places where it's inside of a modal that might need this.